File: python/web_tutorial_mastercopy/wikidata_actor_snippet.py
from bridges.color_grid import *
from bridges.bridges import *
from bridges.color import *
from bridges.data_src_dependent import data_source
import sys
import os

# illustrates retrieving actor-movie data from WikiData

def main():

#if TESTING
    # command line args provide credentials and server to test on
    args = sys.argv[1:]
    bridges = Bridges(int(args[0]), args[1], args[2])
    if len(args) > 3:
        bridges.connector.set_server(args[3])
#else
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")
#endif

    bridges.set_title ("Accessing Wikidata Movie/Actor Data")

    list  = data_source.get_wiki_data_actor_movie(1955, 1955)
    print ("Data Records in 1955: " + str(len(list)) + "\n")
    print ("First 3 records:\n")
    for i in range(3):
        print(str(i) + ")Actor-Movie Data:\n" +  
             "\tMovie: " + list[i].movie_uri + ", " + list[i].movie_name + "\n" +
             "\tActor: " + list[i].actor_uri + ", " + list[i].actor_name + "\n")


if __name__ == "__main__":
    main()

File: java/web_tutorial_mastercopy/wikidata_actor_snippet.java
/**
 * Created by Lucas Estrella on 1/31/2017.
 * lestrell@uncc.edu
 */
import java.util.ArrayList;

import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.ActorMovieWikidata;

// This program fragment illustrates how to access and read the Wikidata actor/movie data

public class wikidata_actor_snippet {

	public static void main(String[] args) throws Exception {

		// note: you must fill in with your Bridges credentials
#if TESTING

		Bridges bridges = new Bridges(Integer.parseInt(args[0]), args[1], args[2]);
		bridges.setServer(args[3]);
#else
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID",
			"YOUR_API_KEY");
#endif
		// set title
		bridges.setTitle("Accessing Wikidata Movie/Actor Data");

		// create data source object
        DataSource ds = bridges.getDataSource();

		// Get a List of Wikidata actor/movie  records from Bridges
		// get the actor movie Wikidata data through the BRIDGES API for 1955.
		// data are available from the early 20th century to now.
		ArrayList<ActorMovieWikidata> list = ds.getWikidataActorMovie(1955, 1955);

		// print out the first record of the dataset
		System.out.println("Actor-Movie Data:\n" +
			"\tMovie: " + list.get(0).getMovieURI()  + " \""  + list.get(0).getMovieName() + "\"\n" +
	     	"\tActor: " +  list.get(0).getActorURI() + " \"" + list.get(0).getActorName() + "\"\n"); 
	}
}
